### PR TITLE
disable stop ride button for 5 seconds on mount

### PR DESCRIPTION
**Pull request by guytepper**

### CODE REVIEW

File: app/screens/route-details/components/live-ride-sheet.tsx [hunk 1]
```diff
@@ -51,24 +51,36 @@ export const LiveRideSheet = observer(function LiveRideSheet(props: { progress;
...
+const StopButton = (props: { loading: boolean } & PressableProps) => {
+  const [isDisabled, setIsDisabled] = useState(true)
+
+  useEffect(() => {
+    // disable the button for 5 seconds
+    // this prevents frequent ride start & stop presses,
+    // and gives the request time to fullfill
+    const timeout = setTimeout(() => {
+      setIsDisabled(false)
+    }, 5000)
```
Comment (planecore) [inline app/screens/route-details/components/live-ride-sheet.tsx:72]: Shouldn't we do that only when the request is loading? There's no need to disable the button for 5 seconds when the request is already fulfilled

**Review by planecore — CHANGES_REQUESTED** [review]

File: app/screens/route-details/components/live-ride-sheet.tsx [hunk 2]
```diff
@@ -61,22 +61,50 @@ const STOP_BUTTON: ViewStyle = {
 }
 
 const StopButton = (props: { loading: boolean } & PressableProps) => {
+  const { loading } = props
   const [isDisabled, setIsDisabled] = useState(true)
```
Comment (planecore) [inline app/screens/route-details/components/live-ride-sheet.tsx:65]: The initial value should be `useState(loading)`

File: app/screens/route-details/components/live-ride-sheet.tsx [hunk 3]
```diff
@@ -61,22 +61,50 @@ const STOP_BUTTON: ViewStyle = {
...
 
   useEffect(() => {
```
Comment (planecore) [inline app/screens/route-details/components/live-ride-sheet.tsx:67]: why are the 2 useEffects needed?

Comment (guytepper) [reply app/screens/route-details/components/live-ride-sheet.tsx:67]: a mistake 🥴

**Review by planecore — APPROVED** [review]

**Pull request merged**